pkg/cache/sql/partition/partition.go

```diff
@@ -0,0 +1,17 @@
+/*
+Package partition represents listing parameters. They can be used to specify which namespaces a caller would like included
+in a response, or which specific objects they are looking for.
+*/
+package partition
+
+import (
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Partition represents filtering of a request's results by namespace or a list of resource names
+type Partition struct {
+	Namespace   string
+	All         bool
```

Comment:
It would be worth commenting each of these items. From what I was reading, it looks like if you don't set All you need to set at least one Name, but that's not documented here.

Reply:
Agreed on comments

Setting All to false and Names to empty set would just mean you get no results from this Partition. Not a particularly useful query I agree - still legit

Reply:
Added comment to function. Yes, as Silvio said that'll work but will result in an empty list.